src/main/java/com/esri/samples/localserver/local_server_feature_layer/LocalServerFeatureLayerSample.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2016 Esri.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.esri.samples.localserver.local_server_feature_layer;
+
+import java.net.URISyntaxException;
+import java.nio.file.Paths;
+
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.geometry.Pos;
+import javafx.scene.Scene;
+import javafx.scene.control.ProgressBar;
+import javafx.scene.layout.StackPane;
+import javafx.stage.Stage;
+
+import com.esri.arcgisruntime.datasource.arcgis.ServiceFeatureTable;
+import com.esri.arcgisruntime.layers.FeatureLayer;
+import com.esri.arcgisruntime.localserver.LocalFeatureService;
+import com.esri.arcgisruntime.localserver.LocalServer;
+import com.esri.arcgisruntime.localserver.LocalServerStatus;
+import com.esri.arcgisruntime.localserver.LocalService.StatusChangedEvent;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.Viewpoint;
+import com.esri.arcgisruntime.mapping.view.MapView;
+
+public class LocalServerFeatureLayerSample extends Application {
+
+  private static final int APPLICATION_WIDTH = 800;
+
+  private ArcGISMap map;
+  private MapView mapView;
+  private LocalServer server;
+  private LocalFeatureService featureService;
+  private ProgressBar featureLayerProgress;
+
+  @Override
+  public void start(Stage stage) throws Exception {
+
+    try {
+      // create stack pane and application scene
+      StackPane stackPane = new StackPane();
+      Scene scene = new Scene(stackPane);
+
+      // set title, size, and add scene to stage
+      stage.setTitle("Local Server Feature Layer");
+      stage.setWidth(APPLICATION_WIDTH);
+      stage.setHeight(700);
+      stage.setScene(scene);
+      stage.show();
+
+      // create a view with a map and basemap
+      map = new ArcGISMap(Basemap.createStreets());
+      mapView = new MapView();
+      mapView.setMap(map);
+
+      // track progress of loading feature layer to map
+      featureLayerProgress = new ProgressBar(-100.0);

```

Comment:
Can you use a `ProgressIndicator` instead like the reverse geocode online sample?